### PR TITLE
Matthewsaum 62275

### PR DESCRIPTION
**Pull request by matthewsaum**
### What does this PR do?
Adds the "dig.X" function to the dig execution module. This allows the resolution of PTR records for entered IP addresses

### What issues does this PR fix or reference?
Fixes: 
[62275](https://github.com/saltstack/salt/issues/62275)

### Previous Behavior
Dig module was unable to resolve PTR records

### New Behavior
Dig module can now resolve PTR records

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by nicholasmhughes** [comment]
re-run pr-windows-2019-x64-py3-pytest

File: salt/modules/dig.py
Comment (nicholasmhughes) [inline salt/modules/dig.py:104]:
```suggestion
    .. versionadded:: 3006.0
```

**Review by nicholasmhughes — CHANGES_REQUESTED** [review]
Looks great! Just one chore to fix. This is probably too late to hit 3005, so it will make the 3006 release.

**Review by nicholasmhughes — DISMISSED** [review]

**Comment by nicholasmhughes** [comment]
re-run pr-photon-3-x86_64-py3-pytest

File: salt/modules/dig.py
Comment (Ch3LL) [inline salt/modules/dig.py:102]: Are you using the name `X` because that is the arg passed to dig?

I think it would make more sense to name the function PTR, since that is the type of record being returned.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Comment by matthewsaum** [comment]
So, this is my first pull request. I'll take any and all feedback as to things I could have/should have done better here. Now that it looks like things are cleaned up and stable for the request. 
And not just for Salt, but in general. My GIT working knowledge is using CI/CD to automate my sysadmin tasks. If ya'll have time to drop a few sentences, it would be awesome. And if not, thanks for helping this along anyway

**Comment by nicholasmhughes** [comment]
re-run pr-centos-7-x86_64-py3-pycryptodome-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-centosstream-9-x86_64-py3-pytest

**Comment by nicholasmhughes** [comment]
@matthewsaum , I think everything went pretty smoothly. The only thing I'd say is to leave the PR alone after it's approved unless there's something big that's wrong. Changing small things, like the newline addition or just merging master where there wouldn't be a conflict, isn't worth dismissing the review and triggering all of the tests again. Thanks for the effort and the contribution!

**Comment by nicholasmhughes** [comment]
re-run pr-photon-3-x86_64-py3-pytest

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**
